docroot/modules/custom/uiowa_core/src/ApiClientBase.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace Drupal\uiowa_core;
+
+use Drupal\Component\Utility\UrlHelper;
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\ClientException;
+use GuzzleHttp\Exception\RequestException;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * A base class for API client services.
+ */
+abstract class ApiClientBase implements ApiClientInterface {
+
+  use StringTranslationTrait;
+
+  /**
+   * The API key for accessing the API.
+   *
+   * @var string|null
+   */
+  protected ?string $apiKey = NULL;
+
+  /**
+   * The length of time the cache.
+   */
+  protected int $cacheLength = 900;
```

Comment:
Wonder if this is too long. If someone is trying to setup the subscribe block, realizes they need a new subscription list, then configures a new one on the dispatch side, it could be 15 min before they see it in the dropdown with no indication that they need to wait for cache to clear.

I think adding the cache to this is a good idea, but mainly to prevent abuse, not to limit use.

<img width="307" alt="Screenshot 2023-09-12 at 11 38 51 AM" src="https://github.com/uiowa/uiowa/assets/4663676/b9756804-7855-4aa8-9ab9-647e1fcf8c49">

Reply:
```suggestion
  protected int $cacheLength = 60;
```

reducing this. if the dispatch team identifies abuse with the calls, we can turn it up. this way, a user can configure, find their way back to the page, switch to layout, add the block and see new stuff pulled.